src/test/java/co/rsk/federate/signing/hsm/client/HSMSigningClientProviderTest.java

```diff
@@ -41,102 +47,57 @@ void getClientV1() throws Exception {
         assertTrue(client instanceof HSMSigningClientV1);
     }
 
-    @Test
-    void getClientV2BTC() throws Exception {
-        HSMClientProtocol protocol = mock(HSMClientProtocol.class);
-        HSMSigningClientProvider clientProvider = new HSMSigningClientProvider(protocol, "BTC");
-        when(protocol.getVersion()).thenReturn(2);
-        HSMSigningClient client = clientProvider.getSigningClient();
-
-        assertTrue(client instanceof PowHSMSigningClientBtc);
+    private static Stream<Arguments> keyIdProvider() {
+        return Stream.of(
+            Arguments.of("BTC", PowHSMSigningClientBtc.class),
+            Arguments.of("RSK", PowHSMSigningClientRskMst.class),
+            Arguments.of("MST", PowHSMSigningClientRskMst.class)
```

Comment:
You can use `PowPegNodeKeyId` enum for these

Reply:
Updated